crates/matrix-sdk/src/media.rs

```diff
@@ -515,6 +519,26 @@ impl Media {
         Ok(content)
     }
 
+    /// Get a media file's content that is only available in the media cache.
+    ///
+    /// # Arguments
+    ///
+    /// * `uri` - The local MXC URI of the media content.
+    async fn get_local_media_content(&self, uri: &MxcUri) -> Result<Vec<u8>> {
+        // Read from the cache.
+        self.client
+            .event_cache_store()
+            .lock()
+            .await?
+            .get_media_content_for_uri(uri)
+            .await?
+            .ok_or_else(|| {
+                // A request to the server would return a 404 NOT_FOUND error, so let's simulate
+                // that.
+                not_found_error("Unknown media ID".to_owned()).into()
```

Comment:
I found this a bit contrived: instead, let's have `get_local_media_content` return `Result<Option<Vec<u8>>>`, and the caller can do something like

```
if let Some(local_uri) = request.source.as_local() {
  match get_local_media_content(local_uri) {
   Err(err) => { /* log the error, and let the function continue */ }
   Ok(Some(val)) => { return Ok(val) }
   Ok(None) => { /* nothing: let the code continue */ }
  }
}
```

Reply:
We don't want the function to continue though. If `get_media_content` continues it will try to load the media request from the cache _again_ and then make a request to the server with a bogus MXC URI.

Reply:
You're right, in that case can it return a more specialized error, and not an HTTP error? No HTTP requests have been done.

Reply:
Done.